src/execution/compiler/operator/sort_translator.cpp

```diff
@@ -200,14 +223,24 @@ void SortTopTranslator::DeclareIterator(FunctionBuilder *builder) {
 }
 
 void SortTopTranslator::GenForLoop(FunctionBuilder *builder) {
+  if (op_->HasLimit() && op_->GetOffset() != 0) {
```

Comment:
You still need a variable if there is **no** limit, but there **is** an offset.

Reply:
We have removed the HasLimit() check.

@17zhangw The op_'s GetOffset() assumes that for the offset to be valid, the limit exists. From the optimizer's perspective, is it safe to remove this check / will you always provide it?

Reply:
This issue can probably be punted to a later time? The optimizer (as I am aware) does not push the limit/offset information down to the `OrderByPlanNode`.

Reply:
Is there a `TERRIER_ASSERT` to be made that'll document our assumptions right now that will yell at us to modify this code if the behavior changes in the future?

Reply:
I think it would make sense to keep the `TERRIER_ASSERT` in the op_'s `GetOffset()` function for now? That way when we start pushing the limit/offset downwards, the assert will at least be tripped if the limit is missing.